### PR TITLE
Functional test: Adding a new model

### PR DESCRIPTION
**Pull request by thofmann**
### Summary

This is the first functional test for the server. It uses the API to test adding a new model.

### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. Check out `new-model-test-TFF-137`
2. `npm install` and `npm test`
3. Verify that the functional tests are running and passing
4. Review the tests for thoroughness and clarity

### Links

- [TFF-132](https://msoese.atlassian.net/browse/TFF-132)

### CODE REVIEW

File: src/store/model-store.js [hunk 1]
```diff
@@ -69,6 +70,15 @@ store.registerHandler('DELETE_ITEM', data => {
 });
 
 store.registerHandler('NEW_MODEL', data => {
+    assert.isString(data.name, 'The model name must be a string');
```
Comment (gearedmonkey) [inline src/store/model-store.js]: Why are we now asserting in main code?

Comment (thofmann) [inline src/store/model-store.js]: It's an assertion library, not just one for testing. This prevents the client from creating an invalid model and ruining the state on the server.

Comment (thofmann) [inline src/store/model-store.js]: Input sanitation, basically.

Comment (thofmann) [inline src/store/model-store.js]: It throws an error which is caught here and returned as a failed API response: https://github.com/TheFourFifths/consus/blob/1727b82d485ec602cb94ebc2d6054e3aa98c632b/src/api/model.js#L33

The response is tested here: https://github.com/TheFourFifths/consus/blob/1727b82d485ec602cb94ebc2d6054e3aa98c632b/test/functional/new-model.js#L65

**Pull request merged**